cfspopcon/formulas/separatrix_conditions/separatrix_operational_space/shared.py

```diff
@@ -0,0 +1,138 @@
+"""Shared functions used for computing the separatrix operational space."""
+import numpy as np
+
+from ....algorithm_class import Algorithm
+from ....unit_handling import Quantity, Unitfull, ureg, wraps_ufunc
+from ...metrics.larmor_radius import calc_larmor_radius
+
+
+def calc_ideal_MHD_wavenumber(beta_e: Unitfull, epsilon_hat: Unitfull, omega_B: Unitfull, tau_i: Unitfull, alpha_t: Unitfull) -> Unitfull:
+    """Calculate k_ideal, which gives the spatial scale of ideal MHD modes.
+
+    Equation G.3 from :cite:`Eich_2021`.
+
+    N.b. G.3 is written in terms of beta_hat = beta_e * epsilon_hat
+    where epsilon_hat = (q R / electron_pressure_decay_length)**2
+    """
+    return np.sqrt(beta_e * epsilon_hat * omega_B**1.5 * (1.0 + tau_i) / alpha_t)
+
+
+def calc_resistive_ballooning_wavenumber(critical_alpha_MHD: Unitfull, alpha_t: Unitfull, omega_B: Unitfull) -> Unitfull:
+    """Calculate k_RBM, which gives the spatial scale of resistive ballooning modes.
+
+    Equation B.12 from :cite:`Eich_2021`
+
+    N.b. There is an extra factor of Lambda_pe multiplied on omega_B compared to
+    table 1. In Appendix A it is noted that Lambda_pe = perpendicular_decay_length / electron_pressure_decay_length = 1.
+    """
+    return np.sqrt(critical_alpha_MHD / alpha_t * np.sqrt(omega_B))
```

Comment:
I've not yet figured out how to get from B.12 to this.

Reply:
Very unclear from the paper, but I've amended the comment to explain why this is written as the square root of the right-hand-side of equation 4 from :cite:`Eich_2021`.